### PR TITLE
Update readme and add RFC template.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,77 @@
...
+  - When consensus about the RFC has been reached the corresponding PR will be
+    approved and the PR merged.
+  - The person that actually merges the PR will create a new *project* for the
+    duration of the implementation (which might span several PRs) in the
+    projects pane of the google-cartographer organization and assign an RFC
+    number.
+  - When you create implementing PRs, please add a link to the RFC in the PR
+    description.
+
+[Cartographer Projects]: https://github.com/googlecartographer/cartographer/projects
```
Comment (jihoonl) [inline README.md]: Why don't we have an organization level project page? as @SirVer suggested [here](https://github.com/googlecartographer/rfcs/pull/1#discussion_r147684523)? You can enable projects for organization under project section in organization section. This [link](https://github.com/organizations/googlecartographer/settings/projects) might work

**Review by jihoonl — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (cschuet) [reply README.md]: Right, that was the idea and I had already updated the text accordingly but forgotten to do the same for the link to the projects page. Fixed now. Thanks.

Comment (jihoonl) [reply README.md]: Oops.. sorry @cschuet, I just found out that [Org project board cannot be public](https://stackoverflow.com/questions/40532661/make-github-organization-projects-public)... The project board is only visible to the organization members.

There is an alternative way of making it public but I am not sure if it is good. See [this post](https://medium.com/@bmpvieira/how-to-make-your-github-organization-project-board-public-fb5a243f28e4)

Comment (cschuet) [reply README.md]: Oh thanks for pointing out I did not know about that. So how about we use the RFCs project pane instead?

Comment (jihoonl) [reply README.md]: I am fine with it. I am actually fine with just using cartographer project page too.

**Review by jihoonl — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (SirVer) [reply README.md]: I think the cartographer project page is most appropriate.

File: 0000-template.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+## Motivation
+[motivation]: #motivation
+
+Why are we doing this? What use cases does it support? What is the expected outcome?
+
+## Approach
+[approach]: #approach
+
+Provide a brief description of the implementation details for the feature. What
```
Comment (SirVer) [inline 0000-template.md]: Reflow to 80 cols? 

Or, as is usually best for prose: One sentence per line.

I am a bit torn here - this is filler text that should be removed, but the RFCs should be written as 1 sentence per line for easier diff reading.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,77 @@
-# rfcs
-RFCs for changes to Google Cartographer projects
+# Cartographer RFCs
+[Cartographer RFCs]: #cartographer-rfcs
+
+For changes more substantial than simple bugfixes and documentation improvements
```
Comment (SirVer) [inline README.md:4]: one sentence per line.

File: 0000-template.md (hunk 2 above)
Comment (cschuet) [reply 0000-template.md]: Yes, let's do the one sentence per line rule in the RFC. I have added a comment in the README.md about that.

File: README.md (hunk 3 above)
Comment (cschuet) [reply README.md:4]: Done

File: README.md (hunk 1 above)
Comment (cschuet) [reply README.md]: Done.

File: 0000-template.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+## Summary
+[summary]: #summary
+
+One paragraph explanation of the feature.
+
+## Motivation
+[motivation]: #motivation
+
+Why are we doing this? What use cases does it support?
```
Comment (SirVer) [inline 0000-template.md]: one sentence per line.

File: README.md (hunk 3 above)
Comment (SirVer) [reply README.md:4]: no?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,80 @@
...
+[When should you write an RFC]: #when-should-you-write-an-rfc
+
+You should provide an RFC if you intent to make a *significant* change or
+contribution to one of Cartographer's repositories. A change is e.g.
+*significant* if it introduces a new or modifies an existing *concept*. The
+process is really very lightweight so we encourage you to err on the side of
+RFCs.
+
+Make sure to have *one sentence per line* in the RFC as it makes reading diffs
+much easier. The 80 characters per line limit does not apply to RFCs.
```
Comment (SirVer) [inline README.md]: drop second sentence. What 80 chars per line limit?

File: 0000-template.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+What is the expected outcome?
+
+## Approach
+[approach]: #approach
+
+Provide a brief description of the implementation details for the feature.
+What existing code paths are you planning to modify.
```
Comment (wohe) [inline 0000-template.md]: End questions with a question mark. Same on the next line.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,80 @@
-# rfcs
-RFCs for changes to Google Cartographer projects
+# Cartographer RFCs
+[Cartographer RFCs]: #cartographer-rfcs
+
+For changes more substantial than simple bugfixes and documentation improvements
+we ask that these be put through a very lightweight design review process.
+
+"RFC" stands for "request for comments" and are used, e.g.
```
Comment (wohe) [inline README.md]: Is this missing a preposition: `e.g. by [rust-lang]`...?

File: README.md [hunk 8]
```diff
@@ -1,2 +1,80 @@
...
+  - [Table of Contents]
+  - [When should you write an RFC]
+  - [Motivation for RFCs]
+  - [Before creating an RFC]
+  - [What the process is]
+
+## When should you write an RFC
+[When should you write an RFC]: #when-should-you-write-an-rfc
+
+You should provide an RFC if you intent to make a *significant* change or
```
Comment (wohe) [inline README.md]: typo: intend?

File: README.md [hunk 9]
```diff
@@ -1,2 +1,80 @@
...
+    approved and the PR merged.
+  - The person that actually merges the PR will create a new *project* for the
+    duration of the implementation (which might span several PRs) in the
+    [projects pane][Cartographer Projects] of the Cartographer repository and
+    assign an RFC number.
+  - When you create implementing PRs, please add a link to the RFC in the PR
+    description.
+
+[Cartographer Projects]: https://github.com/googlecartographer/cartographer/projects
+[RFC repository]: http://github.com/googlecartographer/rfcs
```
Comment (wohe) [inline README.md]: `https`?

**Review by wohe — APPROVED** [review]

File: 0000-template.md (hunk 4 above)
Comment (cschuet) [reply 0000-template.md]: Done

File: 0000-template.md (hunk 6 above)
Comment (cschuet) [reply 0000-template.md]: Done

File: README.md (hunk 3 above)
Comment (cschuet) [reply README.md:4]: Oh this is the README.md. I agree that we want ospl in the RFC template to set an example for future RFC authors. I don't see the benefit for the README.md. Actually I would advise against this practice for the README.md as 
- README.md does not change on a regular basis. Inspection of diffs is therefore less frequent
- Using ospl means you have to use a md formatting viewer to properly enjoy the file.

File: README.md (hunk 7 above)
Comment (cschuet) [reply README.md]: Done

File: README.md (hunk 8 above)
Comment (cschuet) [reply README.md]: Done

File: README.md (hunk 5 above)
Comment (cschuet) [reply README.md]: I am used to having the text files I check into Github formatted at 80 chars per line. This is just to clarify that there is no character line limit for RFCs. I updated the sentence accordingly.

File: README.md (hunk 9 above)
Comment (cschuet) [reply README.md]: Done

File: 0000-template.md [hunk 10]
```diff
@@ -8,15 +8,16 @@ One paragraph explanation of the feature.
...
 What is the expected outcome?
 
 ## Approach
 [approach]: #approach
 
 Provide a brief description of the implementation details for the feature.
-What existing code paths are you planning to modify.
-What new classes or datastructures are you planning to introduce.
+What existing code paths are you planning to modify?
+What new classes or datastructures are you planning to introducer?
```
Comment (wohe) [inline 0000-template.md]: typo: `to introduce`

**Review by wohe — APPROVED** [review]

**Review by SirVer — APPROVED** [review]

**Pull request merged**